docs/developer/README.md

```diff
@@ -0,0 +1,595 @@
+
+#### Requirements
+
+- The functional and error specification for each of the plugin types MUST be defined by the framework
+    - - The data exchange contract SHOULD be versioned to support future changes and compatibility.
+- The plugin SHOULD be an executable/binary that will be executed by the framework with appropriate environment variables and other data through ```stdin```. This data exchange contract MUST be defined in the plugin specification provided by the framework.
+- The initial version of the framework allows plugins to be binaries. Future versions of the framework MAY support plugins that are services invoked through gRPC or HTTP.
+- The plugins MUST be registered in the framework's configuration file. When registered, they can have custom settings that are specific to that plugin. A set of reserved keys like ``name`` will be used by the framework.  The framework MUST pass through other fields, unchanged to the plugin at the time of execution.
+- The location of these plugin binaries MAY be defined as part of registration. However, if not provided, the default path ${HOME}/.ratify/plugins will be used as the path to search for the plugins.
+- The framework MUST validate the existence of plugin binaries in the appropriate paths at the start up and fail with error if any of the registered plugins are not found in the configured paths.
+- [TODO] Permissions and threat model for executing these plugins SHOULD be specified.
+- In addition to the specification, the framework SHOULD provide libraries for writing plugins. A simple CLI for example ```ratify plugin verifier add myverifier``` to create a stub for a plugin using these libraries MAY be provided by the framework.
+
+### Open Questions
+
+- How do we handle version incompatibility issues? Do we need to add a version check API for every plugin that will be used by the framework to determine the compatibility?
+
+## Artifact references
+
+An artifact is defined by a manifest and identified by a reference as per the OCI `{DNS/IP}/{Repository}:[name|digest]`.
+A reference artifact is linked to another artifact through descriptor and enables forming a chain of artifacts as shown below. An artifact that has reference types is termed as ```subject``` against which verification can be triggered using this framework.
+
+![tree view of artifact hierarchy](artifact-hierarchy.svg)
+
+This new way of defining reference types to artifacts and creating relationships is described in the [spec](https://github.com/oras-project/artifacts-spec/releases/tag/1.0.0-draft.1). This framework queries for this graph of reference types artifacts for a subject to verify it. For existing systems that do not support these reference types, there will be storage wrappers that can adapt the corresponding supply chain objects as reference types. The advantage  is to have a uniform data format that can be consumed by different components of the framework.
```

Comment:
should we update this link? https://github.com/oras-project/artifacts-spec/releases/tag/v1.0.0-rc.2